src/queue.js

```diff
@@ -0,0 +1,33 @@
+'use strict'
+
+export default class Queue {
+  constructor() {
+    this._list = []
+    this._backElementPosition = 0
+  }
+  enqueue( element ) {
+    this._list[ this._backElementPosition++ ] = element
+  }
+  dequeue() {
+    let list = this.front()
+    this._list = this._list.reduce( ( memo, element, index ) => {
+      if ( this._list[ index + 1 ] !== undefined ) {
+        memo[ index ] = this._list[ index + 1 ]
+      }
+      return memo
+    }, [] )
+    return list
+  }
+  front() {
+    return this._list[ this.length() - this._backElementPosition ] || null
+  }
```

Comment:
I wonder if `return this._list[0] || null` would work because it seems as if the operation that you're doing should always equal zero for the value of the array's index.